mono/mini/debugger-agent.c

```diff
@@ -10371,8 +10371,8 @@ debugger_thread (void *arg)
 		/* This will break if the socket is closed during shutdown too */
 		if (res != HEADER_LENGTH) {
 			PRINT_DEBUG_MSG (1, "[dbg] transport_recv () returned %d, expected %d.\n", res, HEADER_LENGTH);
-			command_set = (CommandSet)0;
-			command = 0;
+			command_set = CMD_SET_VM;
+			command = CMD_VM_DISPOSE;
```

Comment:
I wonder if we need the same change [here](https://github.com/Unity-Technologies/mono/pull/1679/files#diff-838c3a98ef3566efb4d8deaec7eb7c0853b2416d75de58550f443a5fe1770ab5R10408) as well? This condition will also cause the debugger loop to break and no new agent thread will replace it and no other debugger can attach to the process.  This case is probably much harder to hit, but is worth considering.